# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/base/nsStandardURL.h
# Commit: 9ad299b5161b
# Full Hash: 9ad299b5161b0cedf852d56365310de499f94043
# Author: Valentin Gosu <valentin.gosu@gmail.com>
# Date: 2018-11-14 22:12:17
# Regressor Bug: 1495313
# File Overlap Count: 1
# Description:
#   Bug 1495313 - Allow empty username in URLs r=dragana
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D11254
# ==============================================================================

diff -r 6293789f6e9c -r 9ad299b5161b netwerk/base/nsStandardURL.h
--- a/netwerk/base/nsStandardURL.h	Wed Nov 14 14:27:58 2018 +0000
+++ b/netwerk/base/nsStandardURL.h	Wed Nov 14 15:24:21 2018 +0000
@@ -534,12 +534,18 @@
 nsStandardURL::Userpass(bool includeDelim)
 {
     uint32_t pos=0, len=0;
-    // if there is no username, then there can be no password
-    if (mUsername.mLen > 0) {
-        pos = mUsername.mPos;
-        len = mUsername.mLen;
-        if (mPassword.mLen >= 0)
-            len += (mPassword.mLen + 1);
+    if (mUsername.mLen > 0 || mPassword.mLen > 0) {
+        if (mUsername.mLen > 0) {
+            pos = mUsername.mPos;
+            len = mUsername.mLen;
+            if (mPassword.mLen >= 0) {
+                len += (mPassword.mLen + 1);
+            }
+        } else {
+            pos = mPassword.mPos - 1;
+            len = mPassword.mLen + 1;
+        }
+
         if (includeDelim)
             len++;
     }